tests/manage/pv_services/test_resource_deletion_during_pvc_pod_deletion_and_io.py

```diff
@@ -0,0 +1,326 @@
+
+        pods_deleted = pod_bulk_delete.result()
+        assert pods_deleted, "Deletion of pods failed."
+
+        # Verify pods are deleted
+        for pod_obj in pods_to_delete:
+            pod_obj.ocp.wait_for_delete(pod_obj.name)
+        log.info("Verified: Pods are deleted.")
+
+        # Verify that the mount point is removed from nodes after deleting pod
+        node_pv_mounted = verify_pv_mounted_on_node(node_pv_dict)
+        for node, pvs in node_pv_mounted.items():
+            assert not pvs, (
+                f"PVs {pvs} is still present on node {node} after "
+                f"deleting the pods."
+            )
+        log.info(
+            "Verified: mount points are removed from nodes after deleting "
+            "the pods"
+        )
+
+        pvcs_deleted = pvc_bulk_delete.result()
+        assert pvcs_deleted, "Deletion of PVCs failed."
+
+        # Verify PVCs are deleted
+        for pvc_obj in pvcs_to_delete:
+            pvc_obj.ocp.wait_for_delete(pvc_obj.name)
+        logging.info("Verified: PVCs are deleted.")
+
+        # Verify PVs are deleted
+        for pv_obj in pv_objs:
+            pv_obj.ocp.wait_for_delete(resource_name=pv_obj.name, timeout=300)
+        logging.info("Verified: PVs are deleted.")
+
+        # Verify PV using ceph toolbox. Image/Subvolume should be deleted.
+        for pvc_name, uuid in pvc_uuid_map.items():
+            if interface == constants.CEPHBLOCKPOOL:
+                ret = verify_volume_deleted_in_backend(
+                    interface=interface, image_uuid=uuid,
+                    pool_name=sc_obj.ceph_pool.name
+                )
+            if interface == constants.CEPHFILESYSTEM:
+                ret = verify_volume_deleted_in_backend(
+                    interface=interface, image_uuid=uuid
+                )
+            assert ret, (
+                f"Volume associated with PVC {pvc_name} still exists "
+                f"in backend"
+            )
+
+        log.info("Fetching IO results from the pods.")
+        for pod_obj in io_pods:
+            fio_result = pod_obj.get_fio_results()
+            err_count = fio_result.get('jobs')[0].get('error')
+            assert err_count == 0, (
+                f"FIO error on pod {pod_obj.name}. FIO result: {fio_result}"
+            )
+        log.info("Verified IO result on pods.")
+
+        # Verify number of pods of type 'resource_to_delete'
+        final_num_resource_to_delete = len(pod_functions[resource_to_delete]())
+        assert final_num_resource_to_delete == num_of_resource_to_delete, (
+            f"Total number of {resource_to_delete} pods is not matching with "
+            f"initial value. Total number of pods before deleting a pod: "
+            f"{num_of_resource_to_delete}. Total number of pods present now: "
+            f"{final_num_resource_to_delete}"
+        )
+
+        # Check ceph status
+        ceph_health_check(namespace=config.ENV_DATA['cluster_namespace'])
```

Comment:
can we also add mon,mgr,mds,osd count check at end of the script because here we are deleting ceph pods, to verify we can match output what was it before running this script

Reply:
We are working on verifying the restart count of pods, which will be accompanied by pod count check as well. Tracking this separately [here](https://github.com/red-hat-storage/ocs-ci/issues/382#issuecomment-518155983).